cirq/ops/raw_types_test.py

```diff
@@ -459,6 +459,49 @@ def test_tagged_operation():
     assert op.with_qubits(q2).qubits == (q2,)
 
 
+def test_circuit_diagram():
+    h = cirq.H(cirq.GridQubit(1, 1))
+    tagged_h = h.with_tags('tag1')
+    expected = cirq.CircuitDiagramInfo(wire_symbols=("H['tag1']",),
+                                       exponent=1.0,
+                                       connected=True,
+                                       exponent_qubit_index=None,
+                                       auto_exponent_parens=True)
+    args = cirq.CircuitDiagramInfoArgs(None, None, None, None, None, False)
+    assert cirq.circuit_diagram_info(tagged_h) == expected
+    assert (cirq.circuit_diagram_info(tagged_h,
+                                      args) == cirq.circuit_diagram_info(h))
+
+    c = cirq.Circuit(tagged_h)
+    diagram_with_tags = "(1, 1): ───H['tag1']───"
+    diagram_without_tags = "(1, 1): ───H───"
+    assert str(cirq.Circuit(tagged_h)) == diagram_with_tags
+    assert c.to_text_diagram() == diagram_with_tags
+    assert c.to_text_diagram(include_tags=False) == diagram_without_tags
+
+
+def test_circuit_diagram_corner_cases():
+    global_phase = cirq.GlobalPhaseOperation(coefficient=1.0).with_tags('tag0')
+    assert global_phase._circuit_diagram_info_(args=None) == NotImplemented
+    assert cirq.Circuit(global_phase).to_text_diagram() == ''
```

Comment:
This appears to be two bugs in circuit diagrams.  It should be `'\n\nglobal phase: 1.0'`.  It is `'\n\nglobal phase:'` with an untagged `GlobalPhaseOperation`.  But it is not recognized as a global phase when it is wrapped in `TaggedOperation` and just returns `''`.

Do we want to show tagged global phase operations in a circuit?  How should this work when the circuit combines all global phase ops into a single op?

Reply:
Fixed up the global_phase print outs.  Let me know what you think.